src/cpp/src/whisper/whisper.cpp

```diff
@@ -0,0 +1,207 @@
+// Copyright (C) 2023-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+#include <iostream>
+#include <openvino/openvino.hpp>
+#include <regex>
+#include <thread>
+
+#include "../utils.hpp"
+#include "audio_processing.hpp"
+#include "openvino/genai/audio_utils.hpp"
+#include "openvino/genai/streamer_base.hpp"
+#include "openvino/genai/whisper_generation_config.hpp"
+#include "whisper_models.hpp"
+
+#define AUDIO_CHUNK_SIZE 480000  // 16K sampling X 30 Seconds = 16000 x 30 = 480000
+
+namespace {
+
+void supress_tokens(ov::Tensor& logits, const std::vector<int64_t>& supress_tokens) {
+    auto data = logits.data<float>();
+    for (auto supress_token : supress_tokens) {
+        data[supress_token] = -std::numeric_limits<float>::infinity();
+    }
+}
+
+ov::Tensor encode(ov::InferRequest& request, std::vector<float>& mel_data) {
+    ov::Shape input_shape = {1, 80, 3000};
+    ov::Tensor input_tensor(ov::element::f32, input_shape, mel_data.data());
+
+    request.set_tensor("input_features", input_tensor);
```

Comment:
looks like we need to reset input tensor after `infer` as `request` currently  refers to external `mel_data` data, which can be destroyed later.